tests/helpers/test_network.py

```diff
@@ -26,153 +26,145 @@ async def test_get_url_internal(hass: HomeAssistant):
-    assert _async_get_internal_url(hass, allow_ip=False) == "https://example.local:8123"
-    assert (
-        _async_get_internal_url(hass, require_ssl=True) == "https://example.local:8123"
-    )
+    assert _get_internal_url(hass) == "https://example.local:8123"
+    assert _get_internal_url(hass, allow_ip=False) == "https://example.local:8123"
+    assert _get_internal_url(hass, require_ssl=True) == "https://example.local:8123"
 
     with pytest.raises(NoURLAvailableError):
-        _async_get_internal_url(hass, require_standard_port=True)
+        _get_internal_url(hass, require_standard_port=True)
 
     # Test with internal URL: http://example.local:80/
     await async_process_ha_core_config(
         hass, {"internal_url": "http://example.local:80/"},
     )
 
     assert hass.config.internal_url == "http://example.local:80/"
-    assert _async_get_internal_url(hass) == "http://example.local"
-    assert _async_get_internal_url(hass, allow_ip=False) == "http://example.local"
-    assert (
-        _async_get_internal_url(hass, require_standard_port=True)
-        == "http://example.local"
-    )
+    assert _get_internal_url(hass) == "http://example.local"
+    assert _get_internal_url(hass, allow_ip=False) == "http://example.local"
+    assert _get_internal_url(hass, require_standard_port=True) == "http://example.local"
 
     with pytest.raises(NoURLAvailableError):
-        _async_get_internal_url(hass, require_ssl=True)
+        _get_internal_url(hass, require_ssl=True)
 
     # Test with internal URL: https://example.local:443
     await async_process_ha_core_config(
         hass, {"internal_url": "https://example.local:443"},
     )
 
     assert hass.config.internal_url == "https://example.local:443"
-    assert _async_get_internal_url(hass) == "https://example.local"
-    assert _async_get_internal_url(hass, allow_ip=False) == "https://example.local"
+    assert _get_internal_url(hass) == "https://example.local"
+    assert _get_internal_url(hass, allow_ip=False) == "https://example.local"
     assert (
-        _async_get_internal_url(hass, require_standard_port=True)
-        == "https://example.local"
+        _get_internal_url(hass, require_standard_port=True) == "https://example.local"
     )
-    assert _async_get_internal_url(hass, require_ssl=True) == "https://example.local"
+    assert _get_internal_url(hass, require_ssl=True) == "https://example.local"
 
     # Test with internal URL: https://192.168.0.1
     await async_process_ha_core_config(
         hass, {"internal_url": "https://192.168.0.1"},
     )
 
     assert hass.config.internal_url == "https://192.168.0.1"
-    assert _async_get_internal_url(hass) == "https://192.168.0.1"
-    assert (
-        _async_get_internal_url(hass, require_standard_port=True)
-        == "https://192.168.0.1"
-    )
-    assert _async_get_internal_url(hass, require_ssl=True) == "https://192.168.0.1"
+    assert _get_internal_url(hass) == "https://192.168.0.1"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)